openstack/identity/v2/tenants/requests.go

```diff
@@ -27,3 +27,77 @@ func List(client *gophercloud.ServiceClient, opts *ListOpts) pagination.Pager {
 		return TenantPage{pagination.LinkedPageBase{PageResult: r}}
 	})
 }
+
+// CommonOpts are the parameters that are shared between CreateOpts and UpdateOpts
+type CommonOpts struct {
+	// Name is the name of the project.
+	Name string `json:"name,required"`
```

Comment:
Just a heads up: since it's possible to rename both a user and a project, the `Name` field can't be required, and thus you need two separate sets of options for `Create` and `Update`. Otherwise, yes, they could have been shared :)